File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {

        std::unordered_set<int> num_set(nums.begin(),nums.end());


        int count = 0;
        int longest = 0;
        for(int i = 0;i < nums.size();i++){
            
            if(num_set.count(nums[i]-1) == 0){

                while(num_set.count(nums[i]+count) > 0){
                    count++;
                }

                longest = std::max(longest,count);
                count = 0;

            }

        }

        return longest;



        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    
    // use Set for lookup

    const numsSet = new Set(nums)

    let count = 0
    let sequence = 0
    for(let i = 0;i < nums.length;i++){

        if(!numsSet.has(nums[i]-1)){

            while(numsSet.has(nums[i] + count)){
                count++;
            }


            sequence = Math.max(count, sequence)
            count = 0
        }
    }

    return sequence


};